tests/test_matrix_utils_error_handling.py

```
@@ -0,0 +1,370 @@
+
+        # Mock the nio import to avoid isinstance issues with mocked nio
+        with patch("mmrelay.matrix_utils.isinstance", return_value=False):
+            result = _get_detailed_sync_error_message(mock_response)
+
+        self.assertEqual(result, "Network connectivity issue or server unreachable")
+
+    def test_exception_during_processing(self):
+        """Test handling of exceptions during error message extraction."""
+        mock_response = MagicMock()
+        # Make accessing message raise an exception
+        mock_response.message = property(lambda self: 1 / 0)
```

Ruff (0.12.2)

Unused lambda argument: `self`

(ARG005)